oap-ape/ape-java/ape-common/src/main/java/com/intel/ape/service/netty/NettyMessage.java

```diff
@@ -0,0 +1,491 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.ape.service.netty;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.net.ProtocolException;
+import java.util.function.Consumer;
+
+import com.intel.ape.service.params.ParquetReaderInitParams;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelOutboundHandlerAdapter;
+import io.netty.channel.ChannelOutboundInvoker;
+import io.netty.channel.ChannelPromise;
+import io.netty.handler.codec.LengthFieldBasedFrameDecoder;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * A simple and generic interface to serialize messages to Netty's buffer space.
+ */
+public abstract class NettyMessage {
+    // frame length (4), magic number (4), msg ID (1)
+    static final int FRAME_HEADER_LENGTH = 4 + 4 + 1;

```

Comment:
we can define more variables like `FRAME_LENGTH`, `MAGIC_LENGTH` here.
need consider memory alignment here, like add some padding bytes, align to 2^n bytes ?

Reply:
It's ok to define such variables. Will be added in another pull request.

No need to align since it's just a header, more data will be appended. 
And data is managed by `netty`. It has no such requirement when using its `ByteBuf`.